### ISSUE TITLE
numpy.r_[True, False] is not a boolean array (migrated from Trac #1695)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3246]
Original ticket http://projects.scipy.org/numpy/ticket/1695
Reported 2010-12-06 by trac user O.C., assigned to unknown.

Hello,

I observe the following behavior:

```
numpy.r_[True, False]   -> array([1, 0], dtype=int8)
numpy.r_[True]          -> array([ True], dtype=bool)
```

I would expect the first line to give a boolean array:
array([ True, False], dtype=bool)

This may be a bug.
## 

O.C.
numpy.**version** = '1.4.1'

---

**Comment by thouis** [comment]
Comment in Trac by trac user O.C., 2010-12-06

See also:
http://www.mail-archive.com/numpy-discussionatmention:scipy.org/msg29225.html

**Comment by thouis** [comment]
Attachment in Trac by trac user jpeel, 2011-01-04: [0001-BF-fix-for-find_common_type.patch](http://projects.scipy.org/numpy/attachment/ticket/1695/0001-BF-fix-for-find_common_type.patch)

**Comment by thouis** [comment]
Comment in Trac by trac user jpeel, 2011-01-04

The problem here was actually with the find_common_type function in core/numerictypes.py. This function goes through a typecodes string starting with 'b' which is supposed to be boolean ([http://docs.scipy.org/doc/numpy/reference/arrays.dtypes.html dtype]), but 'b' really means int8. From what I've figured out, '?' is really the typecode for boolean. Putting this at the front of the typecodes string fixes the problem. 

There isn't a problem with numpy._r([True]) or numpy._r([[True],[False]]) because for the first case, find_common_type will just return the dtype of the item if there is only one item and for the second case, find_common_type isn't even used.

**Comment by thouis** [comment]
Comment in Trac by trac user jpeel, 2011-01-04

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-01-23

Looks good to me, needs a test.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-01-23

Fixed in 9b8ff38..173cb74.

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-05-24
